File: NSObject+ValidateJSON.m
//
//  NSObject+ValidateJSON.m
//
//  Created by ArmyXomuiL on 08.05.16.
//  Copyright © 2016 Pirro. All rights reserved.
//

#import "NSObject+ValidateJSON.h"

@implementation NSObject (ValidateJSON)

- (id)nullToNil {
    if ([self isKindOfClass:[NSNull class]]) {
        return nil;
    }
    
    return self;
}

- (id)checkForNull {
    if (![self nullToNil]) {
        return nil;
    }
    
    return self;
}

- (id)checkForNullAndClass:(Class)class {
    if (![self nullToNil]) {
        return nil;
    }
    
    if (![self isKindOfClass:class]) {
        NSLog(@"Wrong class '%@' for value '%@', expected '%@'!", [self class], self ,class);
        return [class new];
    }
    
    return self;
}

- (NSNumber *)convertToNumber {
    NSString *string = (NSString *)self;
    if (![string isKindOfClass:[NSString class]]) return nil;
    
    NSNumberFormatter *formatter = [NSNumberFormatter new];
    formatter.numberStyle = NSNumberFormatterDecimalStyle;
    return [formatter numberFromString:string];
}

@end

File: NSObject+ValidateJSON.swift
//
//  NSObject+ValidateJSON.swift
//
//  Created by ArmyXomuiL on 08.05.16.
//  Copyright © 2016 Pirro. All rights reserved.
//

import Foundation

extension NSObject {

    func nullToNil() -> AnyObject? {
        return self.isKindOfClass(NSNull) ? nil : self
    }
    
    func checkForNull() -> AnyObject? {
        return self.nullToNil() == nil ? nil : self
    }
    
    func checkForNullAndClass(className: AnyClass) -> AnyObject? {
        if self.nullToNil() == nil {
            return nil
        }
        
        if self.isKindOfClass(className) == false {
            print("Wrong class \(self.classForCoder) for value \(self), expected \(className)!")
            return (className as! NSObject.Type).init() as NSObject
        }
        
        return self
    }
    
    func convertToNumber() -> NSNumber? {
        let selfString = String(self)
        
        if selfString.isKindOfClass(NSString) == false {
            return nil
        }
        
        let formatter = NSNumberFormatter()
        formatter.numberStyle = NSNumberFormatterStyle.DecimalStyle
        return formatter.numberFromString(selfString)
    }
    
}